Implement pull request previews using Cloudflare

zregvart opened the pull request:
This implements publishing to preview sites on Cloudflare pages by utilizing the build reusable workflow to generate the site artifacts and uploading them using wrangler, managed by `package.json`. When the preview workflow succeeds, a comment is placed on the pull request pointing to the published URL.

Also includes: [Use reusable workflow for build](https://github.com/enterprise-contract/enterprise-contract.github.io/commit/1396e4a9fb34ac01bdf955e17bbde07a6104e6e7)
This way we can reuse this workflow in publishing from main and from pull requests.

zregvart committed:
Implement pull request previews using Cloudflare

This implements publishing to preview sites on Cloudflare pages by
utilizing the build reusable workflow to generate the site artifacts and
uploading them using wrangler, managed by `package.json`. When the
preview workflow succeeds, a comment is placed on the pull request
pointing to the published URL.

lcarva commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 Red Hat, Inc.
```
Update copyright year?

lcarva commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,104 @@
+      - name: Generate website
+        run: npm --prefix antora ci && npm --prefix antora run build
+
+      - name: Upload artifact
```
I see this workflow can be triggered on pull requests. Will that cause any issues when multiple PRs are uploading artifacts? Not sure if they're properly scoped to the PR. Maybe the `path` below need to have the PR number?

UPDATE: The `preview` workflow seems to indicate that artifacts are properly scoped to a specific run. :tada:

zregvart replied:
Artifacts are property of workflow runs

simonbaird reviewed (approved):
Let's merge and see if it works. :cowboy_hat_face: .

simonbaird commented:
We can see how it goes with #42.

zregvart committed:
Use reusable workflow for build

This way we can reuse this workflow in publishing from main and from
pull requests.

zregvart commented:
There was a typo in the workflow so I pushed https://github.com/enterprise-contract/enterprise-contract.github.io/commit/ec9fdb24c7e878c22d0a7eba2af50c0220975411 directly
